gatsby-node.js

```diff
@@ -42,16 +42,121 @@ const getMessages = (path, language) => {
   }
 }
 
-const outdatedMarkdownPages = [
-  "/dapps/",
-  "/enterprise/",
-  "/eth/",
-  "/learn/",
-  "/wallets/",
-  "/what-is-ethereum/",
-]
-
-exports.onCreateNode = ({ node, getNode, actions }) => {
+/**
+ * Markdown isOutdated check
+ * Parse header ids in markdown file (both translated and english) and compare their info structure.
+ * If this structure is not the same, then the file isOutdated.
+ * If there is not english file, return true
+ * @param {string} path filepath for translated mdx file
+ * @returns boolean for if file is outdated or not
+ */
+const checkIsMdxOutdated = (path) => {
+  const splitPath = path.split("/")
+  splitPath.splice(7, 2)
+  const englishPath = splitPath.join("/")
+
+  const re = /([#]+) [^\{]+\{#([^\}]+)\}/gim
+  let translatedData, englishData
+
+  try {
+    translatedData = fs.readFileSync(path, "utf-8")
+    englishData = fs.readFileSync(englishPath, "utf-8")
+  } catch {
+    console.error("fs.readFileSync error")
+    return true
+  }
+
+  let englishMatch = ""
+  let intlMatch = ""
+  try {
+    englishData.match(re).forEach((match) => {
+      englishMatch += match.replace(re, (_, p1, p2) => p1 + p2)
+    })
+    translatedData.match(re).forEach((match) => {
+      intlMatch += match.replace(re, (_, p1, p2) => p1 + p2)
+    })
+  } catch {
+    console.error("regex error")
+    return true
+  }
+
+  return englishMatch !== intlMatch
+}
+
+/**
+ * JSON isOutdated check
+ * Checks if translation JSON file exists.
+ * If translation file exists, checks that all translations are present (checks keys), and that all the keys are the same.
+ * If translation file exists, isContentEnglish will be false
+ * @param {*} path url path used to derive file path from
+ * @param {*} lang language abbreviation for language path
+ * @returns {{isOutdated: boolean, isContentEnglish: boolean}}
+ */
+const checkIsPageOutdated = async (path, lang) => {
+  // Files that need index appended on the end. Ex page-index.json, page-developers-index.json, page-eth2-index.json
+  const indexFilePaths = ["", "developers", "eth2"]
```

Comment:
Should we just change these file names so we don't have to do this extra logic? Perhaps a question for Luka if that would affect anything in Crowdin?

Reply:
Or I suppose we'd need something like this for the homepage, since it's path is `/`?

Reply:
The other two we could probably get away with taking out the -index portion, but the homepage is the tricky one yeah.